Integrating data (#407)
worldbank/d4di: chapters/data-analysis.tex

then use it as a basis for discussions of how to address data issues during variable construction.
This material will also be valuable during exploratory data analysis.

\section{Constructing analysis datasets}

% What is construction -------------------------------------
The third stage is construction of the dataset you will use for analysis.

> It is at this stage that the raw data is transformed into analysis-ready data, 

Reviewer: ```suggestion
It is at this stage that the clean data is transformed into analysis-ready data, 
```
Data cleaning starts at raw data, variable construction starts with clean data, right?

and understand what units are present in one dataset but not the other,
as well as be able to resolve fuzzy or imperfect matching.
There are some commands such as \texttt{reclink} in Stata
that can provide some useful utilities,
but often a large amount of close examination is necessary
in order to figure out what the matching pattern should be
and how to accomplish it in practice through your code.

In other cases, you will need to join data sources that have different units of observation.
For example, you might be overlaying road location data with household data,
using a spatial match,
or combining school administrative data, such as attendance records and test scores,
with household demographic characteristics from a survey.
Sometimes these cases are conceptually straightforward.
For example, merging a dataset of health care providers
with a dataset of patients comes with a clear linking relation between the two;
the challenge usually occurs in correctly defining statistical aggregations
if the merge is intended to result in a dataset at the provider level.
However, other cases may not be designed with the intention to be merged together,

> such as a dataset of infrastructure access points such as water pumps or schools

Reviewer: ```suggestion
such as a dataset of infrastructure access points, for example, water pumps or schools
```
Two _such as_ in the same sentence

and a dataset of household locations and roads.
In those cases, a key part of the research contribution is figuring out what
a useful way to combine the datasets is.
Since these conceptual constructs are so important
and so easy to imagine different ways to do,
it is especially important that these data integrations are not treated mechanically
and are extensively documented separately from other data construction tasks.

Integrating different datasets may involve changing the structure of the data,
e.g. changing the unit of observation through collapses or reshapes.
This should always be done with great care. 
Two issues to pay extra attention to are missing values and dropped observations.
Merging, reshaping and aggregating data sets can change both the total number of observations
and the number of observations with missing values.
Make sure to read about how each command treats missing observations and,

> whenever possible, add automated checks in the script that throw an error message if the result is changing.

Reviewer: ```suggestion
whenever possible, add automated checks in the script that throw an error message if the result is different than what you expect.
```
How to handle _change_ in unit of obs is what we are talking about here

\subsection{Integrating different data sources}
Often, you will combine or merge information from different data sources together
in order to create the analysis dataset.
For example, you may merge administrative data with survey data
to include demographic information in your analysis,
or you may want to integrate geographic information
in order to construct indicators or controls based on the location of observations.
To do this, you will need to consider the unit of response for each dataset, 
and the identifying variable, to understand how they can be merged.

If the datasets you need to join have the same unit of response, 
merging may be straightforward.
The simplest case is merging datasets at the same unit of response 

> which use a consistent numeric ID.

Reviewer: ```suggestion
which use a consistent, uniquely and fully identifying ID variable. 
```
We have link to what _uniquely and fully identifying_ means earlier in this chapter
